product/roundhouse.tasks/Roundhouse.cs

```diff
@@ -142,12 +153,14 @@ public string RecoveryMode
             set
             {
                 RecoveryMode result;
-                if (!Enum.TryParse(value, true, out result))
+                if (Enum.TryParse(value, true, out result))
                 {
-                    throw new ArgumentOutOfRangeException("value", value, "The value of 'RecoveryMode' must be one of these values: 'NoChange', 'Simple' or 'Full'.");
+                    this.recoveryMode = result;
+                }
+                else
+                {
+                    this.loggingHelper.LogError("The value of 'RecoveryMode' must be one of these values: 'NoChange', 'Simple' or 'Full'. Actual value was '{0}'.", value);

```

Comment:
Look at https://github.com/chucknorris/roundhouse/pull/200/files#diff-0a45691fae4571c0a1ed7b44894a05c2R202 for a convention to setting the error appropriately